python/lsst/meas/algorithms/dynamicDetection.py

```diff
@@ -576,6 +635,332 @@ def _computeBrightDetectionMask(self, maskedImage, convolveResults):
+                    (`lsst.afw.detection.FootprintSet` or `None`).
+                ``negative``
+                    Negative polarity footprints
+                    (`lsst.afw.detection.FootprintSet` or `None`).
+                ``numPos``
+                    Number of footprints in positive or 0 if detection polarity was
+                    negative (`int`).
+                ``numNeg``
+                    Number of footprints in negative or 0 if detection polarity was
+                    positive (`int`).
+                ``background``
+                    Re-estimated background. `None` if
+                    ``reEstimateBackground==False``
+                    (`lsst.afw.math.BackgroundList`).
+                ``factor``
+                    Multiplication factor applied to the configured detection
+                    threshold. (`float`).
+            ``thresholdValue``
+                The final threshold value used to the configure the final round
+                of detection (`float`).
+            ``includeThresholdMultiplier``
+                The final multiplication factor applied to the configured detection
+                threshold. (`float`).
+        """
+        band = "fallback"
+        if exposure.filter is not None:
+            if exposure.filter.hasBandLabel():
+                band = exposure.filter.bandLabel
+        if band in self.config.maxNumPeakPerBand:
+            maxNumPeak = self.config.maxNumPeakPerBand[band]
+        else:
+            maxNumPeak = self.config.maxNumPeakPerBand["fallback"]
+
+        # Set up and configure the adaptive detection task on first iteration.
+        inAdaptiveDetection = True
+        nAdaptiveDetIter = 0
+        thresholdFactor = 1.0
+        if nAdaptiveDetIter == 0:
+            if initialThreshold is None:
+                maxSn = float(np.nanmax(exposure.image.array/np.sqrt(exposure.variance.array)))
+                adaptiveDetThreshold = min(maxSn, 5.0)
+            else:
+                adaptiveDetThreshold = initialThreshold
+            adaptiveDetectionConfig = SourceDetectionConfig()
+            adaptiveDetectionConfig.thresholdValue = adaptiveDetThreshold
+            adaptiveDetectionConfig.includeThresholdMultiplier = initialThresholdMultiplier
+            adaptiveDetectionConfig.reEstimateBackground = False
+            adaptiveDetectionConfig.doTempWideBackground = True
+            adaptiveDetectionConfig.tempWideBackground.binSize = 512
+            adaptiveDetectionConfig.thresholdPolarity = "both"
+            self.log.info("Using adaptive detection with thresholdValue = %.2f and multiplier = %.1f",
+                          adaptiveDetectionConfig.thresholdValue,
+                          adaptiveDetectionConfig.includeThresholdMultiplier)
+            adaptiveDetectionTask = SourceDetectionTask(config=adaptiveDetectionConfig)
+
+        maxNumNegFactor = 1.0
+        while inAdaptiveDetection:
```

Comment:
This loop would be better rewritten as a `for` loop over the maximum iteration count range, with `break` statements replacing `inAdaptiveDetection=False` assignments (I don't think you need that variable at all).  This was done on the `u/jbosch/DM-48966` branch, too, but only after the changes to make the task substitutable for `SourceDetectionTask` (which we don't want to do after all), so I'm not sure it will cherry-pick; might be better to just use it as a reference.